internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -521,35 +492,63 @@ func getScheduleFromAnnotation(
 	return schedule, true
 }
 
-// constructRSCronJob constructs and returns ReclaimSpaceCronJob.
-func constructRSCronJob(name, namespace, schedule, pvcName string) *csiaddonsv1alpha1.ReclaimSpaceCronJob {
+func constructCronJob(jobtype Operation, name, namespace, schedule, pvcName string) client.Object {
```

Comment:
Please write two different `contruct[KR|RS]CronJob` functions.
It's better to avoid unsafe typecasting.